src/App.tsx

```diff
@@ -3,33 +3,56 @@ import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
-type State = {};
+type State = {
+  query: string,
+};
 
 export class App extends React.Component<{}, State> {
-  state: State = {};
+  state: State = {
+    query: '',
+  };
+
+  handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { value } = event.target;
+
+    this.setState({ query: value.toLowerCase() });
+  };
 
   render() {
+    const { query } = this.state;
+
+    const visibleMovies = moviesFromServer.filter(movie => (
+      movie.description.toLowerCase().includes(query)
+      || movie.title.toLowerCase().includes(query)
+    ));
+
     return (
       <div className="page">
         <div className="page-content">
           <div className="box">
             <div className="field">
-              <label htmlFor="search-query" className="label">
-                Search movie
-              </label>
 
               <div className="control">
-                <input
-                  type="text"
-                  id="search-query"
-                  className="input"
-                  placeholder="Type search word"
-                />
+                <label htmlFor="search-query" className="label">
+                  Search movie
+                  <input
+                    type="text"
+                    name="query"
+                    value={this.state.query}
+                    onChange={this.handleChange}
+                    id="search-query"
+                    className="input"
+                    placeholder="Type search word"
+                  />
+                </label>
               </div>
             </div>
           </div>
 
-          <MoviesList movies={moviesFromServer} />
+          <MoviesList movies={!this.state.query
+            ? moviesFromServer
+            : visibleMovies}
+          />
```

Comment:
And here you can use this method
```suggestion
          <MoviesList movies={this.getVisibleMovies()} />
```